visualqa/features.py

```diff
@@ -20,15 +20,10 @@ def get_questions_tensor_timeseries(questions, nlp, max_lenght):
     A numpy ndarray of shape: (nb_samples, timesteps, word_vec_dim)
     '''
     # assert not isinstance(questions, basestring)
-    nb_samples = len(questions)
-    word_vec_dim = nlp(questions[0])[0].vector.shape[0]
-    questions_tensor = np.zeros((nb_samples, max_lenght, word_vec_dim))
-    for i in range(len(questions)):
-        tokens = nlp(questions[i])
-        for j in range(len(tokens)):
-            if j < max_lenght:
-                questions_tensor[i,j,:] = tokens[j].vector
-
+    questions_tensor = np.zeros((len(questions), max_length), dtype='int32')
+    for i, doc in enumerate(questions):
+        for j, token in enumerate(nlp(doc[:max_length])):
```

Comment:
As far as I see we first cut the question here to maximum length and then in the next step search vectors for each word (in case we don't find a word we use the word with id '0') right?
If we would keep only vectors which we know (are in the dictionary) and stop after max_length we would probably have a more informative description for the question? 

E.g: dict = {unknown:0, how:1, many:2, see:3} max_length: 3  Question = "How many elephants are in the image?"
The current approach would generate [1 2 0], my suggestion [1 2 3] so one uninformative 0 less... 

I don't know if this influences the performance since we have a very generous max_lenght anyway but its a suggestion.